vnext/ReactUWP/Views/ReactControl.cpp

```diff
@@ -102,6 +102,12 @@ void ReactControl::AttachRoot() noexcept
   if (!m_touchEventHandler)
     m_touchEventHandler = std::make_shared<TouchEventHandler>(m_reactInstance);
 
+  if (m_previewKeyboardEventHandlerOnRoot)
```

Comment:
It's a weird pattern here, PreviewKeyboardEventHandlerOnRoot is already very smart.

we have this:
class ReactControl : public std::enable_shared_from_this<ReactControl>
so previewKeyboardEventHandlerOnRoot should be shared.
PreviewKeyboardEventHandlerOnRoot has the ability to unhook everything. because previewKeyboardEventHandlerOnRoot is a shared point, to avoid any future problem, I unhook manually here.

Reply:
Well then maybe this ties into my previous question. It wasn't obvious to me where this value was shared. It looked to me like it could be a unique_ptr given how it was used.

You say it's a weird pattern... does it have to be weird? Are you following some existing convention?

Reply:
:enable_shared_from_this is a convention to pass parameters as a callback in another thread.
I removed unhook code here since it's only for precaution and we didn't have any problem yet.


---
In reply to: [295572502](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295572502) [](ancestors = 295572502)